zio-http/jvm/src/main/scala/zio/http/netty/client/NettyClientDriver.scala

```diff
@@ -73,45 +91,31 @@ final case class NettyClientDriver private[netty] (
         }
       }
       .map { jReq =>
-        val closeListener: GenericFutureListener[ChannelFuture] = { (_: ChannelFuture) =>
-          // If onComplete was already set, it means another fiber is already in the process of fulfilling the promises
-          // so we don't need to fulfill `onResponse`
-          nettyRuntime.unsafeRunSync {
-            onComplete.interrupt && onResponse.fail(NettyClientDriver.PrematureChannelClosure)
-          }(Unsafe.unsafe, trace): Unit
-        }
```

Comment:
This existed in order to account for cases that the error is not propagated via the pipeline, e.g., if the server drops the connection unexpectedly or if we errored before reaching the `ClientInboundHandler`.

Have you tested what happens in those cases? Are we still fulfilling the promises or are we risking the application hanging?

Reply:
> This existed in order to account for cases that the error is not propagated via the pipeline, e.g., if the server drops the connection unexpectedly or if we errored before reaching the ClientInboundHandler.

Server dropping a connection results in a `PrematureChannelClosure` exception from Netty.  I have tested this at C1 using Mockserver configured to drop incoming connections.  It even fires correctly in the SSLSpec when there is no valid cert.

Other than that, do you have any other specific cases we can test for?  AFAIK, the Netty pipeline is pretty durable and you can usually depend on the `exceptionCaught` being executed.

Reply:
Additionally, if you interrupt/fail promises in the listener, this occurs before `exceptionCaught` is called and we are then unable to propagate the underlying exception correctly.